### PR TITLE
Add some implementation to TabletMeta

### PR DESCRIPTION
**Pull request by chaoyli**

### CODE REVIEW

**Review by yiguolei — APPROVED** [review]
LGTM

File: be/src/olap/task/engine_clone_task.cpp [hunk 1]
```diff
@@ -586,11 +586,11 @@ OLAPStatus EngineCloneTask::_finish_clone(TabletSharedPtr tablet, const string&
...
         // load src header
-        string clone_header_file = clone_dir + "/" + std::to_string(tablet->tablet_id()) + ".hdr";
-        TabletMeta clone_header(clone_header_file);
-        if ((res = clone_header.load_and_init()) != OLAP_SUCCESS) {
-            OLAP_LOG_WARNING("fail to load src header when clone. [clone_header_file=%s]",
-                             clone_header_file.c_str());
+        string cloned_tablet_meta_file = clone_dir + "/" + std::to_string(tablet->tablet_id()) + ".hdr";
+        TabletMeta cloned_tablet_meta(cloned_tablet_meta_file);
+        if ((res = cloned_tablet_meta.load_and_init()) != OLAP_SUCCESS) {
+            OLAP_LOG_WARNING("fail to load src header when clone. [cloned_tablet_meta_file=%s]",
```
Comment (kangpinghuang) [inline be/src/olap/task/engine_clone_task.cpp]: LOG(WARNING)

File: be/src/olap/task/engine_clone_task.cpp [hunk 2]
```diff
@@ -666,7 +666,7 @@ OLAPStatus EngineCloneTask::_finish_clone(TabletSharedPtr tablet, const string&
 }
 
 
-OLAPStatus EngineCloneTask::_clone_incremental_data(TabletSharedPtr tablet, TabletMeta& clone_header,
+OLAPStatus EngineCloneTask::_clone_incremental_data(TabletSharedPtr tablet, TabletMeta& cloned_tablet_meta,
```
Comment (kangpinghuang) [inline be/src/olap/task/engine_clone_task.cpp]: TabletMeta& cloned_tablet_meta -> TabletMeta* cloned_tablet_met

**Review by kangpinghuang — CHANGES_REQUESTED** [review]

File: be/src/olap/tablet.h [hunk 3]
```diff
@@ -75,10 +75,9 @@ class Tablet : public std::enable_shared_from_this<Tablet> {
     OLAPStatus save_tablet_meta();
 
     void delete_expire_incremental_data();
-    OLAPStatus publish_version(int64_t transaction_id, Version version, VersionHash version_hash);
     const PDelta* get_incremental_delta(Version version) const;
-    OLAPStatus clone_data(const TabletMeta& clone_header,
-                          const std::vector<const PDelta*>& clone_deltas,
+    OLAPStatus clone_data(const TabletMeta& tablet_meta,
+                          const std::vector<RowsetMetaSharedPtr>& clone_deltas,
```
[the remark this answers is not in the document]
Comment (chaoyli) [reply be/src/olap/tablet.h]: I will fix it

File: be/src/olap/task/engine_clone_task.cpp (hunk 1 above)
Comment (chaoyli) [reply be/src/olap/task/engine_clone_task.cpp]: OK

File: be/src/olap/task/engine_clone_task.cpp (hunk 2 above)
Comment (chaoyli) [reply be/src/olap/task/engine_clone_task.cpp]: OK

**Pull request merged**